### ISSUE TITLE
How to do basic FPS-like interaction with a navmesh?

### ISSUE THREAD

**Issue by Elettrotecnica** [c-frame/aframe-extras#343]
Hello,

I am trying to learn how to use navmeshes to constrain the movement of users inside of an environment. For a starter, I am trying to make it work using the WASD controls.

I have read here and there about how to use the nav-mesh component and came out with this [boilerplate](https://abyssinian-cloud-servant.glitch.me)

However, I am not sure whether the castle example I have found in this repository is working as expected: I see a castle, but some grey shapes appear on the ground that I suspect are the navmesh not correctly aligned with the environment (but I am not 100% sure about this...). Furthermore, WASD controls appear to not work when I open the example in the browser.

Is there something I am doing wrong? Where can I find a working example of WASD experience using a navmesh?

Many thanks for any help on this.

**Comment by dirkk0** [comment]
This example should give you a starting point:
https://glitch.com/edit/#!/navmesh-test?path=index.html%3A1%3A0

**Comment by Elettrotecnica** [comment]
Many many thanks for your quick response!

Tweaking your example made me realize there might be something going on with the latest version of aframe-extras I have used in my attempt.

I have remixed your example and turns out that if one replaces the CDN version with the raw github version (which I assume to be the latest) then there won't be any movement.

https://mixolydian-battle-flood.glitch.me

If I use the version from CDN, then everything works as expected. For people having my same problem, I have realized one needs to take care of setting the position of the player using the rig, rather than the camera entity, or else the navmesh will be traslated together with the camera and will be off with respect of the environment.

Thanks a lot again!
